modules/localgov_services_status/src/Plugin/Block/ServiceStatusMessage.php

```diff
@@ -0,0 +1,151 @@
+
+  /**
+   * Drupal\Core\Condition\ConditionManager definition.
+   *
+   * @var \Drupal\Core\Condition\ConditionManager
+   */
+  protected $pluginManagerCondition;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Visible service status nodes.
+   *
+   * @var \Drupal\node\NodeInterface[]
+   */
+  protected $statusNodes = [];
+
+  /**
+   * Cache contexts for visible service statuses.
+   *
+   * @var string[]
+   */
+  protected $cacheContexts = [];
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('plugin.manager.condition'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Service status message block constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Condition\ConditionManager $plugin_manager_condition
+   *   The plugin manager condition service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConditionManager $plugin_manager_condition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->pluginManagerCondition = $plugin_manager_condition;
+    $this->entityTypeManager = $entity_type_manager;
+
+    // Load service status nodes to display.
+    $nids = $this->entityTypeManager->getStorage('node')->getQuery()
+      ->condition('type', 'localgov_services_status')
+      ->condition('status', 1)
+      ->execute();
+    $nodes = $this->entityTypeManager->getStorage('node')->loadMultiple($nids);
+
+    // Check whether service status is visible for given conditions.
+    foreach ($nodes as $node) {
+      if (!$node->get('localgov_service_status_visibile')->isEmpty()) {
+        $conditions = [];
+        $conditions_config = $node->get('localgov_service_status_visibile')->getValue()[0]['conditions'];
+
+        foreach ($conditions_config as $condition_id => $values) {
+          /** @var \Drupal\Core\Condition\ConditionInterface $condition */
+          $condition = $this->pluginManagerCondition->createInstance($condition_id, $values);
+          $conditions[] = $condition;
+          $this->cacheContexts = Cache::mergeContexts($this->cacheContexts, $condition->getCacheContexts());
+        }
+
+        if (ConditionAccessResolver::checkAccess($conditions, 'or')) {
+          $this->statusNodes[] = $node;
+        }
```

Comment:
I am always worried of [heavy lifting within the constructor](https://www.codeproject.com/articles/768052/golden-rules-of-good-oop) :( Perhaps this bit can form a few methods such as@
```
isVisible()
- fetchStatusNodes()
- grabConditions()
- evaluateConditions()
- etc.
```

Reply:
I find creating private methods that are only ever used once for initialising stuff confusing. The purpose of that code block is to initialise the `$statusNodes` and `$cacheContexts` arrays. I don't think that's heavy lifting, but can rewrite it if others feel it needs separating out.

Reply:
I'm don't have a strong opinion, I think I'd might have spun out some of the conditions stuff, but maybe not. 
In the spirit of sharing and comparing. What would have gone though my mind:-

Is it possible anyone might extend it and want something slightly different from the constructor;
Is there anything non-trivial that could be reused and put into a protected or public method (also sometimes trivial code with hard coded values that might want to be changed);
Is the nesting depth getting a bit much so readability would be enhanced by calling some descriptive function names (even if they are only ever called from here).

Reply:
Thanks @ekes @Adnan-cds 

We don't have guidelines for the project on this sort of thing other than Drupal best practices. A quick search through the constructors in Drupal core show they're all fairly simple, but this is often because they're also in a large class hierarchy and each parent is doing something so in the end quite a bit is done in constructors.

Constructors should certainly only be used for initialising stuff and my preference is to do all of this in the constructor when there's no requirement to reuse code. I take the point about readability, but I also find that splitting things out can effect this because, even with the best labelled method and variable names, if you want to actually find out what something is doing you have to inspect the code and constantly jumping around a file checking what's going on is a pain (this is also a problem for object oriented programming in general).

Happy for any suggestions on how to improve this code block.